### PR TITLE
[4.0] Add GDNative Framework support for macOS, add Unix symlink API. 

### PR DESCRIPTION
**Pull request by bruvzg**
Adds ability to select, load and export (both `.dmg` and `.zip`) macOS ".frameworks" as GDNative libraries.
Adds minimal symlink API to `DirAccess` (frameworks usually contain multiple relative symlinks).

Framework export won't work from Windows (If there are symlinks in the framework). NTFS have support for symlinks, but I do not see any way to get relative symlink target path, all functions seems to return full path. And creating symlinks seems to require privilege elevation. Not sure what to do with it (currently it will print a warning).

*Edit:* Relative symlinks are supported and `DeviceIoControl(FSCTL_GET_REPARSE_POINT)` function can get correct relative path, but none of Windows apps seems to use them and all Windows ZIP software extract symlinks as text files with the target path. Even if working `readlink` equivalent is added, Windows users won't be able to acquire macOS framework with correct structure in the first place.

Version for 3.2 - #46860

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]

**Comment by bruvzg** [comment]
Actually, just renaming `64` to `universal` probably won't work and need a bit more changes to the GDNative loading (OS features).

**Comment by bruvzg** [comment]
Removed separate `x86_64` and `arm64`, it's just an unnecessary complication, default export template is universal and there's no way to determine architecture of the custom export template (implementing MachO header reader is easy, but I do not see any benefits of doing it).

File: drivers/windows/dir_access_windows.h [hunk 1]
```diff
@@ -78,6 +78,11 @@ class DirAccessWindows : public DirAccess {
 	virtual Error rename(String p_path, String p_new_path);
 	virtual Error remove(String p_path);
 
+	virtual bool is_link(String p_file) { return false; };
+	virtual String read_link(String p_file) { return p_file; };
+	virtual Error create_link(String p_source, String p_target) { return FAILED; };
+
+	//virtual FileType get_file_type() const;
```
Comment (akien-mga) [inline drivers/windows/dir_access_windows.h:85]: Can be cleaned up.

**Comment by akien-mga** [comment]
> Removed separate `x86_64` and `arm64`, it's just an unnecessary complication, default export template is universal and there's no way to determine architecture of the custom export template (implementing MachO header reader is easy, but I do not see any benefits of doing it).

I guess we'll need to clarify the documentation so users know they can put either a universal binary, an x86_64 or an arm64 binary for the single `64` entry (matching their export template).

File: drivers/windows/dir_access_windows.h (hunk 1 above)
Comment (bruvzg) [reply drivers/windows/dir_access_windows.h]: Done.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**